prefect_utils/snowflake.py

```diff
@@ -194,3 +198,157 @@ def load_ga_data_to_snowflake(
+
+
+@task
+def load_s3_data_to_snowflake(
+    date: str,
+    date_property: str,
+    sf_credentials: dict,
+    sf_database: str,
+    sf_schema: str,
+    sf_table: str,
+    sf_role: str,
+    sf_warehouse: str,
+    sf_storage_integration_name: str,
+    s3_url: str,
+    sf_file_format: str = "TYPE='JSON', STRIP_OUTER_ARRAY=TRUE",
+    files: List[str] = None,
+    pattern: str = None,
+    overwrite: bool = False,
+):
+    """
+    Loads objects in S3 to a generic table in Snowflake, the data is stored in a variant column named
+    `PROPERTIES`. Note that either `files` or `pattern` parameter must be specified.
+    You must explicitly include a separator (/) either at the end of the `s3_url` or at the beginning
+    of each file name specified in the `files` parameter. Typically you would include a trailing (/)
+    in `s3_url`.
+
+    Args:
+      date (str): Date of the data being loaded.
+      date_property (str): Date type property name in the variant `PROPERTIES` column.
+      sf_credentials (dict): Snowflake public key credentials in the format required by create_snowflake_connection.
+      sf_database (str): Name of the destination database.
+      sf_schema (str): Name of the destination schema.
+      sf_table (str): Name of the destination table.
+      sf_role (str): Name of the snowflake role to assume.
+      sf_warehouse (str): Name of the Snowflake warehouse to be used for loading.
+      sf_storage_integration_name (str): Name of the Snowflake storage integration to use. These are
+              configured in Terraform.
+      s3_url (str): Full URL to the S3 path containing the files to load.
+      sf_file_format (str, optional): Snowflake file format for the Stage. Defaults to 'JSON'.
+      files (List[str], optional): File paths relative to `s3_url`.
+      pattern (str, optional): Path pattern/regex to match S3 objects to copy. Defaults to `None`.
+      overwrite (bool, optional): Whether to overwrite existing data for the given date. Defaults to `False`.
+    """
+
+    if not files and not pattern:
+        raise signals.Fail('Either `files` or `pattern` must be specified to run this task.')
+
+    sf_connection = create_snowflake_connection(sf_credentials, sf_role, warehouse=sf_warehouse)
+
+    # Check for data existence for this date
+    try:
+        query = """
+        SELECT 1 FROM {table}
+        WHERE date(PROPERTIES:{date_property})='{date}'
+        """.format(
+            table=qualified_table_name(sf_database, sf_schema, sf_table),
+            date=date,
+            date_property=date_property,
+        )
+        cursor = sf_connection.cursor()
+        cursor.execute(query)
+        row = cursor.fetchone()
+    except snowflake.connector.ProgrammingError as e:
+        if "does not exist" in e.msg:
+            # If so then the query failed because the table doesn't exist.
+            row = None
+        else:
+            raise
+
+    if row and not overwrite:
+        return
```

Comment:
Can we log a message here so we know why this task skipped? Or, since we're using signals in here, raise a SKIP signal, which should make it even more obvious?